docs/source/troubleshooting/tracing.rst

```diff
@@ -55,33 +55,101 @@ Below is the typical sequence to start an *event trace* from an SSH/serial termi
+
+.. code:: bash
+
+   cat /sys/kernel/debug/tracing/trace > trace.txt
+
+Analyzing the results
+=====================
+
+A resulting binary trace file can then be loaded in the front-end GUI tool Kernelshark for off-target analysis.
 
    .. image:: resources/trace-loaded-in-kernelshark.png
 
-``kernelshark`` can be used on a Linux desktop distribution or on Windows with Windows Subsystem for Linux (WSL).
+Kernelshark can be used on a Linux desktop distribution or on Windows using the Windows Subsystem for Linux (WSL).
+
+Kernelshark using Windows Subsystem for Linux (WSL)
+---------------------------------------------------
+
+Kernelshark is a Linux application. You can run it on a native Linux desktop installation or a virtual machine installation. If you would like to run Kernelshark from within a Windows desktop environment, you can do so using the Windows Subsystem for Linux (WSL).
+
+A complete guide to installing and using WSL is outside the scope of this document. However, the following steps show the general process to install and use Kernelshark. For more information, see the `Microsoft documentation <https://learn.microsoft.com/en-us/windows/wsl/install>`_.
+
+#. Install WSL
+
+   From Powershell or Command Prompt run in administrator mode:
+
+   .. code:: console
+
+      wsl --install
+
+   This command installs the default Linux distribution. The remaining instructions assume Ubuntu, but the instructions are similar for other distributions.
+
+#. Restart Windows
+
+#. Launch Ubuntu
+
+   Ubuntu should now be present in your Windows start menu. Select it to launch Ubuntu. This will open a shell window. There will be a delay as Windows downloads and installs your Linux distribution. You will select a username and password for your account in the Linux distribution.
+
+#. Install Kernelshark
+
+   In the shell window, run each of the following commands.
+
+   .. code:: console
+
+      sudo apt update
+      sudo apt install kernelshark
+
+   If you are using a different Linux distribution with a package manager other than ``apt`` the commands will be different.
+
+#. Launch Kernelshark
+
+   Kernelshark should now be present in your Windows start menu. Select it to launch Kernelshark.
+
+   .. note::
+      If appropriate drivers for your graphics hardware are not present in your selected Linux distribution, Kernelshark might open with an empty black window. In this case, you can run Kernelshark with software rendering as a workaround. To do so, run the following command from the Linux shell.
+
+      .. code:: console
+
+         LIBGL_ALWAYS_SOFTWARE=1 kernelshark
+
+#. Copy the trace file
+
+   Use your preferred method to copy the trace file from your target to Windows or to your Linux distribution running in WSL. Common methods include scp, ssh, WebDAV, FTP, etc.
```

Comment:
Would `webdav` have permissions to the trace file?

Reply:
Removed.